### PR TITLE
Simplify Tanimoto kernel.

### PR DESCRIPTION
**Pull request by AustinT**
I made some simplifications to the Tanimoto kernel.

1. Added in my batch Tanimoto similarity code, which also works if for some reason you feed in a vector of all zeros.
2. Remove some redundant logic. For example, there was some stuff about gradients in the Tanimoto code. Tanimoto is a discrete kernel, I think you should never be computing a gradient. Also removed some redundancy in the check for supported metrics.
3. Added an extra option for Tanimoto kernels to support just returning the diagonal (which is just a tensor of ones).

Also ran `black` formatting.

### CODE REVIEW

File: gprotorch/kernels/fingerprint_kernels/base_fingerprint_kernel.py [hunk 1]
```diff
@@ -36,59 +55,38 @@ def _sim(self, x1, x2, postprocess, x1_eq_x2=False, metric='tanimoto'):
...
-            denominator = x1_norm + x2_norm.squeeze(-1) - cross_product
-
-            res = (cross_product/denominator)
-
-            # if x1 and x2 are equal than the Tanimoto similarity is 1
-            if x1_eq_x2 and not x1.requires_grad and not x2.requires_grad:
-                res.diagonal(dim1=-2, dim2=-1).fill_(1)
-
-            # Zero out negative values
+        if metric == "tanimoto":
```
Comment (Ryan-Rhys) [inline gprotorch/kernels/fingerprint_kernels/base_fingerprint_kernel.py:59]: So I guess we'll extend this with elif statements to support the other bit vector distances?

File: gprotorch/kernels/fingerprint_kernels/base_fingerprint_kernel.py [hunk 2]
```diff
@@ -36,59 +55,38 @@ def _sim(self, x1, x2, postprocess, x1_eq_x2=False, metric='tanimoto'):
...
+            res = batch_tanimoto_sim(x1, x2)
             res.clamp_min_(0)
```
Comment (Ryan-Rhys) [inline gprotorch/kernels/fingerprint_kernels/base_fingerprint_kernel.py:61]: Should we have the "zero out negative values" comment above this?

File: gprotorch/kernels/fingerprint_kernels/tanimoto_kernel.py [hunk 3]
```diff
@@ -3,43 +3,56 @@
...
     def __init__(self, **kwargs):
         super(TanimotoKernel, self).__init__(**kwargs)
-        self.metric = 'tanimoto'
-
-    def forward(self, x1, x2, **params):
-        return self.covar_dist(x1, x2, **params)
+        self.metric = "tanimoto"
+
+    def forward(self, x1, x2, diag=False, **params):
+        if diag:
```
Comment (Ryan-Rhys) [inline gprotorch/kernels/fingerprint_kernels/tanimoto_kernel.py:52]: Great stuff! I was wondering what the use cases for returning the diagonal are? I know it's a feature in GPyTorch kernels but haven't seen any examples

File: gprotorch/kernels/fingerprint_kernels/tanimoto_kernel.py [hunk 4]
```diff
@@ -3,43 +3,56 @@
 RDKit fragment features.
 """
 
+import torch
 import gpytorch
 
-from gprotorch.kernels.fingerprint_kernels.base_fingerprint_kernel import BitKernel
+from gprotorch.kernels.fingerprint_kernels.base_fingerprint_kernel import (
```
Comment (Ryan-Rhys) [inline gprotorch/kernels/fingerprint_kernels/tanimoto_kernel.py:9]: What do the brackets do in the import statement?

**Comment by Ryan-Rhys** [comment]
Looks great! Just wondering if the tests in test_tanimoto_kernel.py are passing?

File: gprotorch/kernels/fingerprint_kernels/base_fingerprint_kernel.py (hunk 1 above)
Comment (AustinT) [reply gprotorch/kernels/fingerprint_kernels/base_fingerprint_kernel.py:59]: That was my thought!

File: gprotorch/kernels/fingerprint_kernels/base_fingerprint_kernel.py (hunk 2 above)
Comment (AustinT) [reply gprotorch/kernels/fingerprint_kernels/base_fingerprint_kernel.py:61]: Good catch, that's a mistake. I'll push a fix.

File: gprotorch/kernels/fingerprint_kernels/tanimoto_kernel.py (hunk 3 above)
Comment (AustinT) [reply gprotorch/kernels/fingerprint_kernels/tanimoto_kernel.py:52]: It's used for example in sparse GP regression. I originally didn't have it but this caused my sparse GP implementation to fail.

File: gprotorch/kernels/fingerprint_kernels/tanimoto_kernel.py (hunk 4 above)
Comment (AustinT) [reply gprotorch/kernels/fingerprint_kernels/tanimoto_kernel.py:9]: This was put there by black. It allows imports to be put over multiple lines.

File: gprotorch/kernels/fingerprint_kernels/base_fingerprint_kernel.py (hunk 2 above)
Comment (AustinT) [reply gprotorch/kernels/fingerprint_kernels/base_fingerprint_kernel.py]: Pushed.

**Comment by AustinT** [comment]
> Looks great! Just wondering if the tests in test_tanimoto_kernel.py are passing?

I wasn't sure how to run the tests so I'm not sure...

**Comment by Ryan-Rhys** [comment]
@AustinT Yep I guess we should work on a test runner. For the time being running pytest in test_tanimoto_kernel.py should be good!
